src/app/options/options.component.ts

```diff
@@ -35,13 +35,28 @@ export class OptionsComponent {
       });
     }),
   );
+  submitting$: Observable<boolean> = of(false);
+  @ViewChild('submitButton', {read: ElementRef}) submitButton: ElementRef;
 
-  constructor(private readonly service: SubmitService) {}
-}
+  constructor(readonly service: DataService) {}
 
-function validateFormat(control: AbstractControl): {[key: string]: any} | null {
-  if (!FORMATS.includes(control.value)) {
-    return {invalidFormat: true};
+  ngAfterViewInit() {
+    this.submitting$ = fromEvent(this.submitButton.nativeElement, 'click').pipe(
+      mergeMap(() => {
+        return concat(
+          of(true),
+          of(true).pipe(this.service.submit(), mapTo(false))
```

Comment:
Would it make more sense to take out the input/output control checks from the html and make it part of why submitting would be false?

So if outputControl.invalid or invalidControl.invalid, then submitting is false.